lib/cqm/converter/hds_record.rb

```diff
@@ -163,8 +158,33 @@ def generate_qdm_data_element(dc_fixed_keys, dc_type)
           QDM::FacilityLocation.new.from_json(facility.to_json)
         end
       end
-
+      populate_codesystem_oid(data_element)
+      # iterate over all of the fields
+      # if array or one of the nested types (facility location) dive deeper
+      # if object contains codeSystem and codeSystemOid doesn't have a value then set codeSystemOid to the correct OID
```

Comment:
I think these comments should be right above the `populate_codesystem_oid` function definition.